File: C++/BinaryNumberTriangle.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    void BinaryNumberTriangle(int n) {
        for (int i = 0; i < n; i++) {
            int start = 1;
            if (i % 2 == 0) start = 1;
            else start = 0;
            for (int j = 0; j <= i; j++) {
                cout << start << " ";
                start = 1 - start;
            }
            cout << endl;
        }
    }
};

int main() {
    int n;
    cin >> n;
    Solution obj;
    obj.BinaryNumberTriangle(n);
    return 0;
}

File: Java/BinaryNumberTriangle.java
import java.util.*;

public class BinaryNumberTriangle {
    public static void PrintPattern(int n) {
        for (int i = 0; i < n; i++) {
            int start = (i % 2 == 0) ? 1 : 0;
            for (int j = 0; j <= i; j++) {
                System.out.print(start + " ");
                start = 1 - start;
            }
            System.out.println();
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        PrintPattern(n);
        sc.close();
    }
}